src/app/shared/issue/assignee/assignee.component.ts

```diff
@@ -52,25 +53,33 @@ export class AssigneeComponent implements OnInit {
 
   updateAssignee(): void {
     const newIssue = this.issue.clone(this.phaseService.currentPhase);
+    const oldAssignees = newIssue.assignees;
     newIssue.assignees = this.assignees;
-    this.issueService.updateIssue(newIssue).subscribe(
-      (updatedIssue: Issue) => {
-        this.issueUpdated.emit(updatedIssue);
-      },
-      (error) => {
-        this.errorHandlingService.handleError(error);
-      }
-    );
-    // Update assignees of duplicate issues
     this.issueService
-      .getDuplicateIssuesFor(this.issue)
-      .pipe(first())
-      .subscribe((issues: Issue[]) => {
-        issues.forEach((issue: Issue) => {
-          const newDuplicateIssue = issue.clone(this.phaseService.currentPhase);
-          newDuplicateIssue.assignees = this.assignees;
-          this.issueService.updateIssue(newDuplicateIssue);
-        });
+      .updateIssueWithAssigneeCheck(newIssue)
+      .pipe(
+        switchMap((updatedIssue: Issue) => {
+          this.issueUpdated.emit(updatedIssue);
+          // Update assignees of duplicate issues
+          return this.issueService.getDuplicateIssuesFor(this.issue);
+        }),
+        first(),
```

Comment:
I think that it is important to include the `first()` call here. If you look into the implementation of `IssueService.getDuplicateIssuesFor(issue)`, the Observable does not terminate after returning the duplicate issues. 

This can cause issues like `HttpError: Unexpected end of JSON input` (status 500) and `HttpError: You have exceeded a secondary rate limit. Please wait a few minutes before you try again.`. Maybe there is a better way to implement the method to get duplicate issues...

Reply:
From my tests, it seems like `first()` does not prevent continuous calls (look at Network data after updating assignees) @seanlowjk and I will look into this. Let's keep this PR on icebox for now.